src/sfincs_bmi.cxx

```diff
@@ -11,239 +11,349 @@
 
 #include <cstring>
 #include <iostream>
+
 #include "sfincs_bmi.hxx"
 
 // sfincs_bmi.f90 implements/exports the following bmi functions:
-extern "C" int initialize(char *c_config_file);
-extern "C" int update(double dt);  // Doesn't seem to update current time
+extern "C" int initialize(const char *c_config_file);
 extern "C" int finalize();
-extern "C" void get_start_time(double *tstart);
-extern "C" void get_end_time(double *tend);
-extern "C" void get_time_step(double *dt);
-extern "C" void get_current_time(double *tcurrent);
-extern "C" void get_var_type(const char *c_var_name, char *c_type);
+
+extern "C" int get_component_name(char *name);
+
+extern "C" int update();
+extern "C" int update_until(double t);
```

Comment:
The [BMI spec](https://bmi.readthedocs.io/en/stable/#update-until) expects the target time, I think what's implemented now is the `dt`, i.e. target time - current time. I suppose we can work around that.